### ISSUE TITLE
Adding new column to existing table brokes maxwell

### ISSUE THREAD

**Issue by pasali** [zendesk/maxwell#1166]
I added a new column to my existing table and maxwell stop working with below exception:

    05:23:55,580 DEBUG SchemaChange - SQL_PARSE <- "alter table foo.bar add column baz longtext"    
    05:23:55,661 ERROR MysqlParserListener - (alter_partition_specification add column baz longtext)     
    05:23:55,661 ERROR SchemaChange - Error parsing SQL: 'alter table foo.bar add column baz longtext' 
    05:23:55,661 ERROR AbstractSchemaStore - Error on bin log position Position[BinlogPosition[mysql-bin-changelog.002909:23227468], lastHeartbeat=1543274110491]
    com.zendesk.maxwell.schema.ddl.MaxwellSQLSyntaxError: column at 
        com.zendesk.maxwell.schema.ddl.MysqlParserListener.visitErrorNode(MysqlParserListener.java:87)
              at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:41)
              at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
              at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
              at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
              at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
              at com.zendesk.maxwell.schema.ddl.SchemaChange.parseSQL(SchemaChange.java:88)
              at com.zendesk.maxwell.schema.ddl.SchemaChange.parse(SchemaChange.java:100)
              at com.zendesk.maxwell.schema.AbstractSchemaStore.resolveSQL(AbstractSchemaStore.java:49)
              at com.zendesk.maxwell.schema.MysqlSchemaStore.processSQL(MysqlSchemaStore.java:99)
              at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:243)
              at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:260)
              at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getRow(BinlogConnectorReplicator.java:494)
              at com.zendesk.maxwell.replication.BinlogConnectorReplicator.work(BinlogConnectorReplicator.java:150)
              at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:27)
              at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:224)
              at com.zendesk.maxwell.Maxwell.start(Maxwell.java:156)
              at com.zendesk.maxwell.Maxwell.main(Maxwell.java:245)

**Comment by osheroff** [comment]
do you know what the original SQL was?  I'm trying to see how maxwell could have parsed all that into `alter_partition_specification`, it really doesn't make too much sense to me....

**Comment by pasali** [comment]
hi, @osheroff  sorry for the delay. Original SQL was ````alter table foo.bar modify column baz datetime(6);````

**Comment by osheroff** [comment]
how on earth is that getting parsed as an `alter_partition_specification`?  ummmmmmm..... what's the actual name of the database and table?

**Comment by pasali** [comment]
database name is specific so it cant be the problem. Table name is `order_status` so this might be problem.

**Comment by osheroff** [comment]
please just give me the SQL unchanged.  I can't see how the SQL you gave me could possibly cause that error.

**Comment by pasali** [comment]
@osheroff  here is another example 

```
12:51:29,297 DEBUG SchemaChange - SQL_PARSE <- "alter table main.order add column completed bit(1) not null default true after price"
12:51:29,372 ERROR MysqlParserListener - (alter_partition_specification add column completed bit ( 1 ) not null default true after price)
12:51:29,372 ERROR SchemaChange - Error parsing SQL: 'alter table main.order add column completed bit(1) not null default true after price'
12:51:29,372 ERROR AbstractSchemaStore - Error on bin log position Position[BinlogPosition[mysql-bin-changelog.005964:6119154], lastHeartbeat=1551182659631]
com.zendesk.maxwell.schema.ddl.MaxwellSQLSyntaxError: column
	at com.zendesk.maxwell.schema.ddl.MysqlParserListener.visitErrorNode(MysqlParserListener.java:87)
	at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:41)
	at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
	at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
	at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
	at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
	at com.zendesk.maxwell.schema.ddl.SchemaChange.parseSQL(SchemaChange.java:88)
	at com.zendesk.maxwell.schema.ddl.SchemaChange.parse(SchemaChange.java:100)
	at com.zendesk.maxwell.schema.AbstractSchemaStore.resolveSQL(AbstractSchemaStore.java:49)
	at com.zendesk.maxwell.schema.MysqlSchemaStore.processSQL(MysqlSchemaStore.java:99)
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:243)
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:260)
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getRow(BinlogConnectorReplicator.java:494)
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.work(BinlogConnectorReplicator.java:150)
	at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:27)
	at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:224)
	at com.zendesk.maxwell.Maxwell.start(Maxwell.java:156)
	at com.zendesk.maxwell.Maxwell.main(Maxwell.java:245)
`

**Comment by haeken** [comment]
I think "order" is a reserved word for mysql and antlr can't parse this correctly. The solution is updating this part as main.\`order\`. I tested it and it seems to be working.

**Comment by osheroff** [comment]
yup @haeken's got it right.  Stock mysql doesn't allow `order` as a table name but presumably aurora does (thx, amazon).  I can make the exception in the parser to allow it.

**Comment by pjebs** [comment]
@osheroff double check the aurora assumption.

**Comment by pjebs** [comment]
It looks like stock mysql can have tables called `order`: https://stackoverflow.com/questions/3137910/how-can-i-make-a-table-in-mysql-called-order

You just have to wrap it in ticks when querying.

**Comment by osheroff** [comment]
sure.  anything in quotes is permissable.  But some keywords are allowed
unquoted, too, see
https://github.com/zendesk/maxwell/blob/master/src/main/antlr4/imports/generate_tokens.rb#L281
for all the shennanigans.

**Comment by osheroff** [comment]
the issue seems to indicate that somehow some SQL with "order" as a table
name got through the SQL server with the table-name unquoted.  So I'm just
guessing, but it seems that aurora allows this where AFAIK mysql doesn't.

**Comment by haeken** [comment]
@osheroff  we tested your claim, but it doesn't seem be an aurora problem. Mysql also allows this query. So can you make fix for this problem?

**Comment by pasali** [comment]
@osheroff  thanks for the fix. When do you planning to release new version ?

**Comment by osheroff** [comment]
fixed in 1.21.0

**Comment by pasali** [comment]
@osheroff sorry for bothering but when will docker image be ready for last release ?
